### ISSUE TITLE
addon does not work with reverse proxy or assigning a port on the host

### ISSUE THREAD

**Issue by omriasta** [hassio-addons/addon-vscode#25]
# Problem/Motivation

> New Ingress not working with reverse proxy and port assignment also fails

## Expected behavior

> Should work normally

## Actual behavior

>  Black screen when trying using the reverse proxy and the Open Web-UI button, 403 Forbidden when using the port assigned to host for the addon

## Steps to reproduce

> Setup nginx reverse proxy and create a host for hassio or create an entry using the forwarded port

## Proposed changes

> (If you have a proposed change, workaround or fix,
> describe the rationale behind it)

Here is the log file, I only removed the IP of my accessing browser and replaced with X.X.X.X:

**UPDATE: Also added the log for the 403 forbidden error as well as the config I have:

[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-banner.sh: executing... 

-----------------------------------------------------------
 Hass.io Add-on: Visual Studio Code
 Fully featured VSCode experience, to edit your HA config in the browser, including auto-completion!
-----------------------------------------------------------
 Add-on version: 0.3.0
 You are running the latest version of this add-on.
 System: Ubuntu 18.10  (amd64 / qemux86-64)
 Home Assistant version: 0.91.3
 Supervisor version: 159
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] code-server.sh: executing... 
[cont-init.d] code-server.sh: exited 0.
[cont-init.d] nginx.sh: executing... 
[cont-init.d] nginx.sh: exited 0.
[cont-init.d] user.sh: executing... 
[cont-init.d] user.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[13:38:07] INFO: Starting the code server...
INFO  code-server v1.792-vsc1.33.1
INFO  Additional documentation: http://github.com/codercom/code-server
INFO  Initializing {"data-dir":"/data/vscode","extensions-dir":"/data/vscode/extensions","working-dir":"/config","log-dir":"/root/.cache/code-server/logs/20190415173809295"}
INFO  Starting webserver... {"host":"127.0.0.1","port":"8443"}
WARN  No certificate specified. This could be insecure.
WARN  Documentation on securing your setup: https://github.com/codercom/code-server/blob/master/doc/security/ssl.md
WARN  Launched without authentication.
INFO   
INFO  Started (click the link below to open):
INFO  http://localhost:8443/
INFO   
INFO  Starting shared process [1/5]...
[13:38:09] INFO: Starting NGinx...
INFO  Connected to shared process
[15/Apr/2019:13:39:13 -0400] 200 X.X.X.X, 172.17.0.3, 172.30.32.1(172.30.32.2) GET / HTTP/1.1 (Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[15/Apr/2019:13:39:13 -0400] 200 X.X.X.X, 172.17.0.3, 172.30.32.1(172.30.32.2) GET /1-f2dd3d.bundle.js HTTP/1.1 (Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[15/Apr/2019:13:39:13 -0400] 200 X.X.X.X, 172.17.0.3, 172.30.32.1(172.30.32.2) GET /main.css HTTP/1.1 (Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[15/Apr/2019:13:39:13 -0400] 200 X.X.X.X, 172.17.0.3, 172.30.32.1(172.30.32.2) GET /f2dd3d.bundle.js HTTP/1.1 (Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
[15/Apr/2019:13:39:24 -0400] 101 X.X.X.X, 172.17.0.3, 172.30.32.1(172.30.32.2) GET / HTTP/1.1 (Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
INFO  WebSocket opened / {"client":1}
INFO  WebSocket closed / {"client":1,"code":1006}
INFO  WebSocket opened / {"client":2,"ip":"127.0.0.1"}
[15/Apr/2019:13:39:25 -0400] 101 X.X.X.X, 172.17.0.3, 172.30.32.1(172.30.32.2) GET / HTTP/1.1 (Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
INFO  WebSocket closed / {"client":2,"code":1006}
INFO  WebSocket opened / {"client":3,"ip":"127.0.0.1"}
[15/Apr/2019:13:39:25 -0400] 101 X.X.X.X, 172.17.0.3, 172.30.32.1(172.30.32.2) GET / HTTP/1.1 (Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
INFO  WebSocket closed / {"client":3,"code":1006}
INFO  WebSocket opened / {"client":4,"ip":"127.0.0.1"}
[15/Apr/2019:13:39:25 -0400] 101 X.X.X.X, 172.17.0.3, 172.30.32.1(172.30.32.2) GET / HTTP/1.1 (Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
INFO  WebSocket closed / {"client":4,"code":1006}
INFO  WebSocket opened / {"client":5,"ip":"127.0.0.1"}
2019/04/15 13:45:43 [error] 449#449: *57 access forbidden by rule, client: 172.30.32.1, server: a0d7b954-vscode, request: "GET / HTTP/1.1", host: "XXXXXX.duckdns.org:443"
[15/Apr/2019:13:45:43 -0400] 403 X.X.X.X(172.30.32.1) GET / HTTP/1.1 (Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)
2019/04/15 13:46:05 [error] 449#449: *58 access forbidden by rule, client: 172.30.32.1, server: a0d7b954-vscode, request: "GET / HTTP/1.1", host: "XXXXXXXX.duckdns.org:443"
[15/Apr/2019:13:46:05 -0400] 403 X.X.X.X(172.30.32.1) GET / HTTP/1.1 (Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36)

CONFIG:
{
  "ssl": false,
  "certfile": "fullchain.pem",
  "keyfile": "privkey.pem",
  "packages": [],
  "init_commands": []
}

**Comment by Fma965** [comment]
i also have the same issue

**Comment by riemers** [comment]
Black screen too, cleared all application data. Didn't help.

**Comment by omriasta** [comment]
Not sure if this helps but if I bypass the proxy completely and use http://local_ip:port to access hassio, the addon does load. So I assume the issue is the URL being used to load. I am getting errors connecting on this URL in chrome dev console:
wss://hass.my_name.duckdns.org/api/hassio_ingress/U-xZYdWzuU2kA3bDtDxBQUl7XvM4jGVx6dQXuAUElZQ/

**Comment by Fma965** [comment]
yes it is definitely the reverse proxy,

**Comment by rafuz** [comment]
Same here, from reverse proxy no way to load it in Ingress.
This is the system log:

```
19-04-15 22:36:11 ERROR (MainThread) [aiohttp.server] Unhandled exception
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/aiohttp/web_protocol.py", line 448, in start
    await resp.write_eof()
  File "/usr/local/lib/python3.7/site-packages/aiohttp/web_response.py", line 444, in write_eof
    await self._payload_writer.write_eof(data)
  File "/usr/local/lib/python3.7/site-packages/aiohttp/http_writer.py", line 138, in write_eof
    self._write(chunk)
  File "/usr/local/lib/python3.7/site-packages/aiohttp/http_writer.py", line 67, in _write
    raise ConnectionResetError('Cannot write to closing transport')
ConnectionResetError: Cannot write to closing transport
```

**Comment by garywilson4** [comment]
same issue here too

**Comment by omriasta** [comment]
My guess is that this file needs to be amended to include all local networks (192.168.0.0/16):
https://github.com/hassio-addons/addon-vscode/blob/master/vscode/rootfs/etc/nginx/servers/ingress.conf

**Comment by omriasta** [comment]
Looks like the issue is fixed for me in 0.91.4

**Comment by Fma965** [comment]
Can confirm, this is working on 0.91.4
